src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -76,7 +76,11 @@ const useFetchChannelHook = () => {
 
     try {
       const channelList = ChannelList.fromChannelAPI(channel, type[channelType]);
-      await channelList.saveIfLatest(localDb);
+      if (channel?.type === 'topics') {
+        await channelList.save(localDb);
+      } else {
+        await channelList.saveIfLatest(localDb);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a conditional check for the channel type and calls different save methods accordingly. However, it's not clear what the difference is between `save` and `saveIfLatest`. If `saveIfLatest` only saves if the data is newer, then it might be better to always use this method to prevent overwriting newer data with older data. If there's a specific reason to use `save` for 'topics' type, please add a comment explaining why.

```diff
-      if (channel?.type === 'topics') {
-        await channelList.save(localDb);
-      } else {
-        await channelList.saveIfLatest(localDb);
-      }
+      await channelList.saveIfLatest(localDb);
```

Also, consider improving error handling in the catch block. Currently, it just logs a generic message which doesn't provide much context about the error. It would be more helpful to log the actual error message or even rethrow the error after logging so that it can be handled upstream.

```diff
-      console.log('error on helperChannelPromiseBuilder');
+      console.error('Error on helperChannelPromiseBuilder:', e);
+      throw e;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->